core/deployment/src/main/java/io/quarkus/deployment/jvm/JvmModulesReconfigurer.java

```diff
@@ -0,0 +1,58 @@
+package io.quarkus.deployment.jvm;
+
+import java.util.List;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.deployment.builditem.ModuleOpenBuildItem;
+
+/**
+ * Interface for reconfiguring JVM module restrictions on the running JVM.
+ * It's an interface as I expect us to possibly explore different strategies
+ * to accomplish this.
+ */
+public interface JvmModulesReconfigurer {
```

Comment:
This seems like it's not well suited for the deployment module. Maybe `runtime-dev` would be more appropriate?